src/main/java/hsleiden/stenentijdperk/stenentijdperk/Controllers/BoardController.java

```diff
@@ -54,11 +50,13 @@ public StaticHut getHut(int stapel) {
     }
 
     public void onResourceButtonClick(int index, int input) {
-        if (!boardmodel.getPlaced() && boardmodel.requestCap(index) - boardmodel.requestVillagers(index) != 0
-                && playercontroller.getPositie(boardmodel.getPlayer(), index) == 0) {
-            // Dit veranderd de hoeveelheid stamleden van een speler
-            boardmodel.decreaseVillagers(index, input);
-            plaatsenStamleden(index, input);
+        if(FirebaseController.getBoard(String.valueOf(localPlayer.getLobby())).getPlayer() == localPlayer) {
+            if (!boardmodel.getPlaced() && boardmodel.requestCap(index) - boardmodel.requestVillagers(index) != 0
+                    && playercontroller.getPositie(boardmodel.getPlayer(), index) == 0) {
+                // Dit veranderd de hoeveelheid stamleden van een speler
+                boardmodel.decreaseVillagers(index, input);
+                plaatsenStamleden(index, input);
+            }
```

Comment:
De code ziet er goed ui. Nou weet ik niet precies wat FirebaseController.getBoard doet, maar het lijkt er op alsof je elke keer dat je een knop klikt een call doet naar firebase. Dit is heel onhandig, want op die manier ga je heel veel calls maken die onnodig zijn.